src/main/java/com/github/webicitybrowser/spec/fetch/connection/imp/FetchConnectionImp.java

```diff
@@ -0,0 +1,58 @@
+package com.github.webicitybrowser.spec.fetch.connection.imp;
+
+import com.github.webicitybrowser.spec.fetch.Body;
+import com.github.webicitybrowser.spec.fetch.FetchRequest;
+import com.github.webicitybrowser.spec.fetch.FetchResponse;
+import com.github.webicitybrowser.spec.fetch.connection.FetchConnection;
+import com.github.webicitybrowser.spec.fetch.connection.FetchConnectionInfo;
+import com.github.webicitybrowser.spec.fetch.imp.FetchResponseImp;
+import com.github.webicitybrowser.spec.http.HTTPRequest;
+import com.github.webicitybrowser.spec.http.HTTPService;
+import com.github.webicitybrowser.spec.http.response.HTTPResponse;
+import com.github.webicitybrowser.spec.http.response.HTTPSuccessResponse;
+import com.github.webicitybrowser.webicity.core.net.ProtocolContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStreamReader;
+
+public class FetchConnectionImp implements FetchConnection {
+
+	private final FetchConnectionInfo info;
+	private final HTTPService httpService;
+	private final static Logger logger = LoggerFactory.getLogger(FetchConnectionImp.class);
+
+	public FetchConnectionImp(FetchConnectionInfo fetchConnectionInfo, HTTPService httpService) {
+		this.info = fetchConnectionInfo;
+		this.httpService = httpService;
+	}
+
+	@Override
+	public FetchConnectionInfo info() {
+		return info;
+	}
+
+	@Override
+	public FetchResponse send(FetchRequest request) {
+		ProtocolContext context = new ProtocolContext(request.method(), redirectURL -> true);
+		HTTPResponse response = null;
+		try {
+			response = httpService.resolveRequest(HTTPRequest.createRequest(request.url(), context));
+		} catch(Exception e) {
+			logger.error(e.getClass().toString());
+			return FetchResponse.createNetworkError();
+		}
+
+		if (response instanceof HTTPSuccessResponse successResponse) {
+			return convertHTTPResponseToFetchResponse(successResponse);
+		} else {
+			logger.error("Unhandled HTTP response object: " + response);
+			return FetchResponse.createNetworkError();
+		}
+	}
+
+	private FetchResponse convertHTTPResponseToFetchResponse(HTTPSuccessResponse response) {
+		return new FetchResponseImp(Body.createBody(new InputStreamReader(response.getInputStream()),  new byte[] {}, -1));
```

Comment:
You put two spaces after `new`. There should only be one.

Reply:
Also i dont know what is going on in my IDE there is one space

Reply:
I guess I can fix it after merge then